doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,924 @@
+* This section describes the cold startup, shutdown, restart and soft reboot. 
+
+### Cold Startup
+#### Definition
+* The device is initially powered down.  Turn on the power and the device boots into its default state. The Switch and all the DPUs will be up or down depending on the configuration
+* Switch: Turning on the power will boot the switch and be ready for traffic
+* DPU: Turning on the power to the DPU will boot and be ready for traffic
+#### Trigger
+* Switch: Power on switch.
+* DPU: Bootup configuration
+#### Use case
+* Switch: Day0 setup, Maintenance, RMA
+* DPU: Day0, Day1-N Provisioning, Maintenance, RMA
+#### DPU cold startup Sequence
+* The chassis is powered up and the host is booting up.
+* The switch PMON is registered with the configDB state change handler.
+* DPU "admin_status: up" is set in the configDB.
+* The switch PMON gets the admin up notification from the configDB
+* The switch PMON invokes the platform API to power on the DPU
+* DPU boots up and attaches itself to the midplane.
+* Once SONiC is up, the state progression is updated for every state transition on the DPU_STATE table in the chassisStateDB
+
+### DPU startup sequence diagram
+<p align="center"><img src="./images/dpu-startup-seq.svg"></p>
+
+### Configuring startup and shutdown
+* The DPUs can be powered down by configuring the admin_status as shown.
+* The corresponding switch configDB table is also shown
+
+#### config_db.json
+CHASSIS_MODULE table holds the list and configuration of DPU modules in a smartswitch chassis. It allows user to administratively bring down a DPU
+```
+{
+    "CHASSIS_MODULE": {
+        "DPU0": {
+            "admin_status": "down"
+        },
+        "DPU1": {
+            "admin_status": "down"
+        }
+    }
+}
+```
+#### switch configDB
+```
+Key: "CHASSIS_MODULE|DPU0"
+    "CHASSIS_MODULE|DPU0": {
+        "value": {
+            "admin_status":"down"
+        }
+    }
+```
+
+### Shutdown
+#### Definition
+* The device will be in powered up state.  Gracefully shutdown and tun off the power
+* Switch: Gracefully shutdown all DPU SONiC turn off DPU power. Gracefully shutdown Switch SONiC and tun off the power to the switch
+* DPU: Gracefully shutdown SONiC and turn off power
+#### Trigger
+* Switch: Turn off the power to the switch
+* DPU: Configuration or CLI
+#### Use case
+* Switch: Maintenance, Critical alarm, RMA
+* DPU: Maintenance, Critical alarm, Service migration, RMA
+#### DPU shutdown sequence
+* There could be two possible sources for DPU shutdown. 1. A configuration change to DPU "admin_status: down" 2. The GNOI logic can trigger it when it fails to communicate with the DPU.
+* The host sends a GNOI signal to gracefully shutdown the DPU.  Upon timeout the host may trigger the switch PMON to shutdown the DPU. NPU-DPU (GNOI) soft reboot workflow will be captured in another document.
```

Comment:
Should we include a segment addressing ungraceful shutdown or boot-up sequences to ensure comprehensive coverage?

Reply:
This is vendor specific.